tests/suites/dashboard/dashboard.sh

```diff
@@ -6,8 +6,7 @@ run_dashboard_deploy() {
 	juju expose dashboard
 	juju relate dashboard controller
 
-	juju wait-for application dashboard
-	sleep 5 # short wait for relation data to update
```

Comment:
Should we keep the sleep? It's kind of ugly but it was to prevent race conditions / eventually consistency problems where the data doesn't update instantly.

Reply:
I asked myself the exact same question and check the wait_for function, which covers this use case:

https://github.com/juju/juju/blob/5399523ddf59d2ae74f16603e7861803397a21c2/tests/includes/wait-for.sh#L41-L43

Reply:
Ah, nice one.